falcon-regression/merlin/src/test/java/org/apache/falcon/regression/SLA/FeedSLAMonitoringTest.java

```diff
@@ -0,0 +1,278 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.falcon.regression.SLA;
+
+import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.entity.v0.Frequency;
+import org.apache.falcon.regression.Entities.FeedMerlin;
+import org.apache.falcon.regression.core.bundle.Bundle;
+import org.apache.falcon.regression.core.helpers.ColoHelper;
+import org.apache.falcon.regression.core.response.ServiceResponse;
+import org.apache.falcon.regression.core.util.HadoopUtil;
+import org.apache.falcon.regression.core.util.InstanceUtil;
+import org.apache.falcon.regression.core.util.TimeUtil;
+import org.apache.falcon.regression.core.util.AssertUtil;
+import org.apache.falcon.regression.core.util.BundleUtil;
+import org.apache.falcon.regression.testHelper.BaseTestClass;
+import org.apache.falcon.resource.SchedulableEntityInstance;
+import org.apache.falcon.resource.SchedulableEntityInstanceResult;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.log4j.Logger;
+import org.joda.time.DateTime;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.Collections;
+
+
+/**
+ * Feed SLA monitoring tests.
+ * Test assumes following properties are set in startup.properties of server :
+ *      *.feed.sla.statusCheck.frequency.seconds=60
+ *      *.feed.sla.lookAheadWindow.millis=60000
+ */
+@Test(groups = "distributed")
+public class FeedSLAMonitoringTest extends BaseTestClass {
+
+    private ColoHelper cluster = servers.get(0);
+    private FileSystem clusterFS = serverFS.get(0);
+    private String baseTestHDFSDir = cleanAndGetTestDir();
+    private String feedInputPath = baseTestHDFSDir + "/input" + MINUTE_DATE_PATTERN;
+    private List<String> slaFeedNames;
+    private List<Frequency> slaFeedFrequencies;
+    private String clusterName;
+    private static final Logger LOGGER = Logger.getLogger(FeedSLAMonitoringTest.class);
+
+    private String startTime;
+    private String endTime;
+    private String slaStartTime;
+    private String slaEndTime;
+    private int noOfFeeds;
+    private int statusCheckFrequency;
+
+    private static final Comparator<SchedulableEntityInstance> DEPENDENCY_COMPARATOR =
+            new Comparator<SchedulableEntityInstance>() {
+                @Override
+                public int compare(SchedulableEntityInstance o1, SchedulableEntityInstance o2) {
+                    int tagDiff = o1.getTags().compareTo(o2.getTags());
+                    if (tagDiff != 0) {
+                        return tagDiff;
+                    }
+                    int clusterDiff = o1.getCluster().compareTo(o2.getCluster());
```

Comment:
I intended to use compareTo but since it tags comparison was missing, overrided it by my method. Anyways I can use compareTo with tags handled explicitly. Will make the changes.